### PR TITLE
python310Packages.mistune: 0.8.4 -> 2.0.4

### PR DESCRIPTION
**Pull request by dotlambda**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/development/python-modules/m2r/default.nix [hunk 1]
```diff
@@ -38,7 +38,7 @@ buildPythonPackage rec {
     homepage = "https://github.com/miyakogi/m2r";
     description = "Markdown to reStructuredText converter";
     license = licenses.mit;
-    maintainers = with maintainers; [ SuperSandro2000 ];
+    maintainers = with maintainers; [ AndersonTorres SuperSandro2000 ];
```
Comment (AndersonTorres) [inline pkgs/development/python-modules/m2r/default.nix:41]: What am I doing here??

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/m2r/default.nix (hunk 1 above)
Comment (dotlambda) [reply pkgs/development/python-modules/m2r/default.nix:41]: You were a maintainer of the removed `pkgs/tools/text/m2r`.

Comment (dotlambda) [reply pkgs/development/python-modules/m2r/default.nix:41]: @AndersonTorres Do you want me to remove your name?

**Comment by smancill** [comment]
Looks like this broke `python3Packages.nbconvert`:

```
python3.10-nbconvert> ERROR: Could not find a version that satisfies the requirement mistune<2,>=0.8.1 (from nbconvert) (from versions: none)
python3.10-nbconvert> ERROR: No matching distribution found for mistune<2,>=0.8.1
```

**Comment by vcunat** [comment]
Did you have any plans for 22.05?

**Comment by risicle** [comment]
Completely forgot about that :/

**Comment by dotlambda** [comment]
It's only a small number of packages that are marked insecure on 22.05, I don't think we need to take action unless issues are opened. We can use `mistune_2_0` where possible though.

**Comment by smancill** [comment]
`jupyter` is broken on 22.05:

https://discourse.nixos.org/t/jupyter-broken-in-22-05/21108/1

**Pull request merged**